pkg/client/events/replay_client_integration_test.go

```diff
@@ -0,0 +1,150 @@
+//go:build integration
+
+package events_test
+
+import (
+	"context"
+	"encoding/json"
+	"fmt"
+	"sync/atomic"
+	"testing"
+	"time"
+
+	"cosmossdk.io/depinject"
+	"github.com/stretchr/testify/require"
+
+	"github.com/pokt-network/poktroll/pkg/client/events"
+	"github.com/pokt-network/poktroll/pkg/observable"
+	"github.com/pokt-network/poktroll/testutil/testclient/testeventsquery"
+)
+
+// Create the generic event type and decoder for the replay client
+
+var _ messageEvent = (*tEvent)(nil)
+
+type messageEvent interface {
+	EventMessage() string
+}
+
+type tEvent struct {
+	Message string `json:"message"`
+}
+
+type messageEventReplayObs observable.ReplayObservable[messageEvent]
+
+func (t *tEvent) EventMessage() string {
+	return t.Message
+}
+
+func newDecodeEventMessageFn() events.NewEventsFn[messageEvent] {
+	return func(eventBz []byte) (messageEvent, error) {
+		t := new(tEvent)
+		if err := json.Unmarshal(eventBz, t); err != nil {
+			return nil, err
+		}
+		if t.Message == "" {
+			return nil, events.ErrEventsUnmarshalEvent
+		}
+		return t, nil
+	}
+}
+
+// newMessageEventBz returns a new message event in JSON format
+func newMessageEventBz(eventNum int32) []byte {
+	return []byte(fmt.Sprintf(`{"message":"message_%d"}`, eventNum))
+}
+
+func TestReplayClient_Remapping(t *testing.T) {
+	var (
+		ctx              = context.Background()
+		connClosed       atomic.Bool
+		delayEvent       atomic.Bool
```

Comment:
I think the name is the opposite of what it actually is.
```suggestion
		firstEventDelayed       atomic.Bool
```